compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -2070,6 +2126,9 @@ public void visit(BLangRecordLiteral recordLiteral) {
                     if (names.contains(name)) {
                         this.dlog.error(keyExpr.pos, DiagnosticCode.DUPLICATE_KEY_IN_RECORD_LITERAL,
                                         recordLiteral.expectedType.getKind().typeName(), name);
+                    } else if (noOfInclusiveTypes > 0 && !neverTypedKeys.contains(name)) {
+                        this.dlog.error(keyExpr.pos, DiagnosticCode.DUPLICATE_KEY_IN_RECORD_LITERAL,
```

Comment:
Isn't this a possible duplication? Not definite, right?

Reply:
How about a scenario like this?
```ballerina
type Foo record {|
    int i;
    int...;
|};

public function main() {
    Foo f = {i: 3, "k": 1};
    map<int> m = {...f, k: 2};
}
```

Reply:
The error we currently have is "invalid usage of {0} literal: duplicate key ''{1}''".

What I meant is, `f` here may or may not have a field with key `k`. So saying duplicate key here may not always be correct, right?

Shall we say something along the lines of "possible duplicate"?

Reply:
Updating your other error a bit, how about something like

`invalid usage of mapping constructor expression: spread field ''{0}'' may have already specified key ''{1}''`

Reply:
I have changed to `invalid usage of mapping constructor expression: spread field ''{0}'' may have already specified keys`. Seems like it is appropriate.